vet-service/src/test/java/com/petclinic/vet/presentationlayer/VetControllerUnitTest.java

```diff
@@ -262,6 +273,34 @@ void getTopThreeVetsWithTheHighestRating() {
         Mockito.verify(ratingService, times(1))
                 .getTopThreeVetsWithHighestAverageRating();
 
+    }
+    @Test
+    void getRatingBasedOnDate() {
+
+
+        String exisingYearDate="2023";
+
+        Map<String, String> ratingQueryParams = new HashMap<>();
+        ratingQueryParams.put("year", exisingYearDate);
+
+        when(ratingService.getRatingsOfAVetBasedOnDate(vet.getVetId(),ratingQueryParams))
+                .thenReturn(Flux.just(ratingDateResponseDTO,ratingDateResponseDTO2));
+        client.get()
+                .uri("/vets/"+VET_ID+"/ratings/date?year={year}",exisingYearDate)
+                .accept(MediaType.APPLICATION_JSON)
+                .exchange()
+                .expectStatus().isOk()
+                .expectHeader().contentType(MediaType.APPLICATION_JSON)
+                .expectBodyList(RatingResponseDTO.class)
+                .value(ratingDTOs ->{
+                            assertEquals(2, ratingDTOs.size());
```

Comment:
Could assert the values in the ratingDTOs list in a later bug, but works well for now